src/session.ts

```diff
@@ -24,29 +25,39 @@ export class DebugSession implements IDebugger.ISession {
   }
 
   /**
-   * The client session to connect to a debugger.
+   * Whether the debug session is disposed.
    */
-  private _client: IClientSession;
-
-  get id() {
-    return this.client.name;
+  get isDisposed(): boolean {
+    return this._isDisposed;
   }
 
-  get type() {
-    return this.client.type;
+  /**
+   * A signal emitted when the debug session is disposed.
+   */
+  get disposed(): ISignal<this, void> {
+    return this._disposed;
   }
 
+  /**
+   * Returns the API client session to connect to a debugger.
+   */
   get client(): IClientSession {
     return this._client;
   }
 
+  /**
+   * Sets the API client session to connect to a debugger to
+   * the given parameter.
+   *
+   * @param client - The new API client session.
+   */
   set client(client: IClientSession | null) {
     if (this._client === client) {
       return;
     }
 
     if (this._client) {
-      Signal.clearData(this._client);
+      this._client.iopubMessage.disconnect(this._handleEvent, this);
```

Comment:
And `Signal.clearData(this._client.iopubMessage)` would not be suitable because it would clear too much?

Reply:
Yes, we don't want to remove other signals, only the one of the debugger session.

Reply:
Indeed, we don't want to remove any other slot, only the one of this session.

Reply:
Good, thanks for the confirmation :+1: